plugins/catalog-react/src/components/EntityTagPicker/EntityTagPicker.tsx

```diff
@@ -94,9 +94,12 @@ export const EntityTagPicker = (props: EntityTagPickerProps) => {
 
   useEffect(() => {
     updateFilters({
-      tags: selectedTags.length ? new EntityTagFilter(selectedTags) : undefined,
+      tags:
+        selectedTags.length && Object.keys(availableTags ?? {}).length
+          ? new EntityTagFilter(selectedTags)
+          : undefined,
```

Comment:
I feel this ternary is becoming a bit hard to see what's going on at a glance - would it be possible to break it out?

Reply:
updated it! let me know if that is better